src/frontend/web_application/src/components/Dropdown/index.jsx

```diff
@@ -1,145 +1,171 @@
     className: null,
-    alignement: null,
   };
   WithDropdownControl.displayName = `WithDropdownControl(${WrappedComponent.displayName || WrappedComponent.name || 'Component'})`;
 
   return WithDropdownControl;
 };
 
+
 class Dropdown extends Component {
+
   static propTypes = {
     id: PropTypes.string.isRequired,
+    alignRight: PropTypes.bool, // force align right
+    children: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
     className: PropTypes.string,
-    position: PropTypes.oneOf(['bottom', 'left']),
-    closeOnClick: PropTypes.bool,
-    closeOnClickExceptSelectors: PropTypes.arrayOf(PropTypes.string),
-    show: PropTypes.bool,
-    children: PropTypes.node,
+    closeOnClick: PropTypes.bool, // should Dropdown close on click?
+    closeOnClickExceptSelectors: PropTypes.arrayOf(PropTypes.shape({})),
+    closeOnScroll: PropTypes.bool, // should Dropdown close on windows scroll ?
+    isMenu: PropTypes.bool,
+    position: PropTypes.oneOf(['top', 'bottom']),
     onToggle: PropTypes.func,
+    show: PropTypes.bool,
   };
 
   static defaultProps = {
+    alignRight: false,
+    children: null,
     className: null,
-    position: null,
     closeOnClick: false,
     closeOnClickExceptSelectors: null,
+    closeOnScroll: false,
+    position: 'bottom',
+    isMenu: false,
+    onToggle: str => str,
     show: false,
-    children: undefined,
-    onToggle: null,
   };
 
-  componentDidMount() {
-    this.$dropdown = jQuery(`#${this.props.id}`);
-    // eslint-disable-next-line no-new
-    new Foundation.Dropdown(this.$dropdown, {
-      hOffset: 0,
-      vOffset: 0,
-    });
-    // `positionClass` option (eg. 'right bottom') is always overriden
-    // use css classes instead (cf. alignement and position)
-    // see https://github.com/zurb/foundation-sites/pull/9019
+  state = {
+    isOpen: false,
+    offset: {},
+  };
 
-    if (this.onToggle) {
-      this.$dropdown.on('show.zf.dropdown', () => {
-        this.onToggle(true);
-      });
-      this.$dropdown.on('hide.zf.dropdown', () => {
-        this.onToggle(false);
-      });
-    }
+  componentDidMount() {
+    this.dropdownControl = document.getElementById(`${CONTROL_PREFIX}-${this.props.id}`);
 
-    if (this.props.closeOnClickExceptSelectors) {
-      this.handleDocumentClick = (ev) => {
-        if (!this.props.show) {
-          return;
-        }
+    this.handleDocumentScroll = () => { this.toggle(false); };
 
-        const target = ev.target;
+    this.handleDocumentClick = (ev) => {
+      const target = ev.target;
+      const exceptSelectors = this.props.closeOnClickExceptSelectors;
 
-        if (this.$dropdown.is(target) || this.$dropdown.find(target).length) {
-          return;
-        }
+      const dropdownClick = !this.props.closeOnClick &&
+        (this.dropdown === target || this.dropdown.contains(target));
 
+      const exeptSelectorsClick = exceptSelectors &&
+        exceptSelectors.find(selector => (selector === target));
```

Comment:
I'm not sure this works : target is not a string or may be selector is not a cssSelector but something more complicated.
(it looks like something a bit magic)